### DISCUSSION TITLE
What is the point of this switch?

### DISCUSSION THREAD

**Discussion by retrievaaaa** [nothings/stb#1340] [Q&A] [answered]
I dont really see the point of this switch:
https://github.com/nothings/stb/blob/af1a5bc352164740c1cc1354942b1c6b72eacb8a/stb_truetype.h#L4026-L4062

The only difference is the divisor, which is the same as kernel_width - so why the switch? 
Even in the default case, its just the kernel_width as divisor.

**Comment by PathogenDavid** [comment, marked as answer]
As stated in the comment immediately above the switch, it's an optimization:

https://github.com/nothings/stb/blob/af1a5bc352164740c1cc1354942b1c6b72eacb8a/stb_truetype.h#L4025

**Reply by nothings** [reply]
Yes. Specifically, compilers can turn division by a constant integer into multiplication by a constant integer, which is much faster.
